src/builtins.c

```diff
@@ -778,6 +778,32 @@ JL_CALLABLE(jl_f__call_in_world)
     return ret;
 }
 
+JL_CALLABLE(jl_f__call_in_world_nonpure)
```

Comment:
Care to explain why do we need this?

Reply:
If you use regular `call_in_world` from inference, which is a pure context, it won't actually apply the world age update, because of the context. This call lets you temporarily switch to nonpure context to do the evaluation.